### PR TITLE
PWN-8453 - Striga API. get_user_details

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8453

## Description of Work
- some changes are not changes, but just moving files

### CODE REVIEW

File: solana/src/main/java/org/p2p/solanaj/utils/TweetNaclFast.java [hunk 1]
```diff
@@ -760,12 +767,17 @@ public Signature(byte [] theirPublicKey, byte [] mySecretKey) {
...
+        @NonNull
+        public byte [] detached(byte [] message) throws SignFailed {
+            try {
+                byte[] signedMsg = this.sign(message);
+                byte[] sig = new byte[signatureLength];
+                for (int i = 0; i < sig.length; i++)
+                    sig[i] = signedMsg[i];
+                return sig;
+            } catch (Exception signError) {
+                throw new SignFailed(signError);
```
Comment (gslevinkov) [inline solana/src/main/java/org/p2p/solanaj/utils/TweetNaclFast.java:779]: better then SomeUnknownToGodException

File: app/src/main/res/values/environment_config.xml [hunk 2]
```diff
@@ -41,5 +41,7 @@
     <string name="googleClientId">553127941597-j88bgo2invnpiefq2j1fhq8duju6o2tq.apps.googleusercontent.com</string>
 
     <string name="alarmErrorsServiceBaseUrl">https://oncall.keyapp.org/</string>
+
+    <string name="strigaProxyServiceBaseUrl">https://payment.keyapp.org/striga/api</string>
```
Comment (eduardmaximovich) [inline app/src/main/res/values/environment_config.xml:45]: missing ending slash

**Review by eduardmaximovich — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/StrigaProxyApiInterceptor.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private fun generateHeaderSignature(): String {
+        val currentUnixTime = Date().time.toString()
+
+        val signedUnixTime = TweetNaclFast.Signature(byteArrayOf(), tokenKeyProvider.keyPair.copyOf())
+            .detached(currentUnixTime.toByteArray())
+            .toBase64Instance()
+            .base64Value
+
+        return "$currentUnixTime:$signedUnixTime".also {
+            println(tokenKeyProvider.publicKey)
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/StrigaProxyApiInterceptor.kt:40]: I think there is no need to print this

File: app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/StrigaProxyApiInterceptor.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            .addHeader("Signed-Message", generateHeaderSignature())
+            .build()
+
+        return chain.proceed(request)
+    }
+
+    /**
+     * @return format: unix_time:unix_time_signed_by_private_key
+     */
+    private fun generateHeaderSignature(): String {
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/interceptor/StrigaProxyApiInterceptor.kt:31]: I think we need a class for this signature, to make test coverage for this fun

File: app/src/main/java/org/p2p/wallet/striga/StrigaUserIdProvider.kt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package org.p2p.wallet.striga
+
+class StrigaUserIdProvider {
+    fun getUserId(): String {
+        return "ed0a1e3a-3e7c-4ef8-b8e2-6fbb7fa44d68"
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/striga/StrigaUserIdProvider.kt:5]: Left TODO here, to don't forget change or remove this x D

Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/striga/StrigaUserIdProvider.kt:5]: and by the way, is this sensible information ?

File: app/src/main/java/org/p2p/wallet/striga/signup/repository/StrigaSignupDataDatabaseRepository.kt [hunk 6]
```diff
@@ -56,22 +65,12 @@ class StrigaSignupDataDatabaseRepository(
         dao.updateOrInsertData(entity)
         success()
     } catch (error: Throwable) {
-        handleError(error)
+        StrigaDataLayerError.from(
```
Comment (Davran331122) [inline app/src/main/java/org/p2p/wallet/striga/signup/repository/StrigaSignupDataDatabaseRepository.kt:68]: I think we can make ext. `StrigaDataLayerError.asFailureResult(error,default)`

**Review by Davran331122 — APPROVED** [review]

**Pull request merged**